### PR TITLE
DR | Update & add keyboard-only e2e tests

### PR DESCRIPTION
**Pull request by Mottie**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
  > - Update HLR keyboard-only test to make it more stable
  > - Re-enable NOD keyboard-only test
  > - Add SC keyboard-only test
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/91696

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Decision Reviews: HLR, NOD & Supplemental Claims

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/995/tests/995-keyboard-only.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+      cy.url().should('include', CONTACT_INFO_PATH);
+      cy.tabToElement('button.usa-button-primary[id$="continueButton"]');
+      cy.realPress('Space');
+
+      // *** Primary phone radios
+      cy.url().should(
+        'include',
+        chapters.infoPages.pages.choosePrimaryPhone.path,
+      );
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/995/tests/995-keyboard-only.cypress.spec.js:75]: All of these eslint messages are for the addition of a `cy.wait(100)` (100 ms wait) needed to allow focus to be set on the page header before attempting to tab down the page to the targeted element

File: src/platform/utilities/ui/focus.js [hunk 2]
```diff
@@ -66,9 +95,12 @@ export function focusElement(selectorOrElement, options, root) {
  *  component we're waiting for (could be an element in shadow DOM)
  * @example waitForRenderThenFocus('h3', document.querySelector('va-radio').shadowRoot);
  */
-const isCypressRunning = typeof Cypress !== 'undefined' && Cypress.env('CI');
+const isCypressRunning =
+  typeof Cypress !== 'undefined' &&
+  (Cypress.env('CI') || environment.isLocalhost());
```
Comment (shaunburdick) [inline src/platform/utilities/ui/focus.js:100]: I don't think `environment.isLocalhost()` means Cypress is running?
I would add an additional `isLocalHost = environment.isLocalhost();` constant, then adjust the statements below to do something like `const defaultTime = isCypressRunning || isLocalhost ? 0 : 250;`

File: src/applications/appeals/10182/tests/10182-keyboard-only.cypress.spec.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ import cypressSetup from '../../shared/tests/cypress.setup';
 // Seems that keyboard only tests using FireFox and Electron
 // fail while Chrome works fine. Further investigation is needed to resolve
 // https://dsva.slack.com/archives/CBU0KDSB1/p1708717681733839
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/10182/tests/10182-keyboard-only.cypress.spec.js:15]: You're right!

File: src/applications/appeals/shared/tests/cypress.helpers.js [hunk 4]
```diff
@@ -1,15 +1,16 @@
 import { parseDateWithOffset } from '../utils/dates';
 import { SELECTED } from '../constants';
 
-export const getRandomDate = () =>
+export const getRandomDate = (offset = Math.random() * 6) =>
   parseDateWithOffset({
-    months: -Math.floor(Math.random() * 6 + 1),
-    days: -Math.floor(Math.random() * 10),
+    months: -(3 + offset),
+    days: -offset,
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/shared/tests/cypress.helpers.js:7]: Derp, good catch!

File: src/platform/utilities/ui/focus.js (hunk 2 above)
Comment (Mottie) [reply src/platform/utilities/ui/focus.js:100]: We only want to reduce the defaultTime when Cypress is running locally or in CI. If I only use `isLocalHost` then when I manually test the pages locally (outside of Cypress), this function may not focus on a header inside a web component. Having the focus on the header doesn't matter as much when Cypress is running tests, especially keyboard-only tests, at full speed.

Comment (shaunburdick) [reply src/platform/utilities/ui/focus.js:100]: In that case I suggest we rename `isCypressRunning` to something more declarative to what it's doing?

Comment (Mottie) [reply src/platform/utilities/ui/focus.js:100]: Done, renamed to `noAsyncFocusWhenCypressRunningInCiOrLocally`

Comment (Mottie) [reply src/platform/utilities/ui/focus.js:100]: I also removed the code setting the `maxIteration` to `1` since that change doesn't matter anymore

**Review by anniebtran — DISMISSED** [review]
LGTM

**Review by anniebtran — APPROVED** [review]

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**